python/taichi/lang/expr.py

```diff
@@ -243,13 +243,22 @@ def atomic_add(self, other):
     return ti.expr_init(taichi_lang_core.expr_atomic_add(self.ptr, other_ptr))
 
   def __pow__(self, power, modulo=None):
-    assert isinstance(power, int) and power >= 0
+    if not isinstance(power, int) or abs(power) > 12:
+      return Expr(taichi_lang_core.expr_pow(self.ptr, Expr(power).ptr))
     if power == 0:
       return Expr(1)
-    ret = self
-    for i in range(power - 1):
-      ret = ret * self
-    return ret
+    negative = power < 0
+    tmp = self
+    ret = 1
+    while power:
+      if power & 1:
+        ret *= tmp
+      tmp *= tmp
```

Comment:
and this line to `tmp = ti.expr_init(tmp * tmp)`?

Reply:
It works!